### PR TITLE
isort coordinates

### PR DESCRIPTION
**Pull request by nstarman**
Signed-off-by: Nathaniel Starkman (@nstarman) <nstarkman@protonmail.com>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Review by mhvk — APPROVED** [review]
I obviously did not look in detail, but the auto sorting certainly seemed reasonable for the spot checks that I did. So, happy to merge if tests pass.

File: astropy/coordinates/angle_formats.py [hunk 1]
```diff
@@ -22,12 +22,13 @@
...
-                     IllegalMinuteWarning, IllegalMinuteError,
-                     IllegalSecondWarning, IllegalSecondError)
+from astropy import units as u
 from astropy.utils import format_exception, parsing
 from astropy.utils.decorators import deprecated
-from astropy import units as u
+
+from .errors import (
+    IllegalHourError, IllegalHourWarning, IllegalMinuteError, IllegalMinuteWarning,
+    IllegalSecondError, IllegalSecondWarning)
```
Comment (eerovaher) [inline astropy/coordinates/angle_formats.py:31]: Should we make this `black`-compliant?

File: astropy/coordinates/builtin_frames/icrs_cirs_transforms.py [hunk 2]
```diff
@@ -8,24 +8,18 @@
...
 from astropy.coordinates.representation import (
-    SphericalRepresentation,
-    CartesianRepresentation,
-    UnitSphericalRepresentation,
-    CartesianDifferential,
-)
+    CartesianDifferential, CartesianRepresentation, SphericalRepresentation,
+    UnitSphericalRepresentation)
+from astropy.coordinates.transformations import (
+    AffineTransform, FunctionTransformWithFiniteDifference)
```
Comment (eerovaher) [inline astropy/coordinates/builtin_frames/icrs_cirs_transforms.py:15]: The previous code was `black`-compliant, so if the new code was `black`-compliant too then the diffs here would be smaller.

File: astropy/coordinates/angle_formats.py (hunk 1 above)
Comment (mhvk) [reply astropy/coordinates/angle_formats.py:31]: Fight of the auto tools... But it does bring up a good point that perhaps this should be done as part of what seems likely to be an upcoming black conversion...

Comment (nstarman) [reply astropy/coordinates/angle_formats.py:31]: isort has a black-compliant setting. When the black PR is rolled out, isort will be changed then and will re-sort these commits.
The reason that it's probably better to do isort now and not bundled with black is that import order sometimes matters (see ``coordinates/builtin_frames.__init__.py``).
So my suggestions to just leave this as is and let black do its thing later.

File: astropy/coordinates/builtin_frames/icrs_cirs_transforms.py (hunk 2 above)
Comment (nstarman) [reply astropy/coordinates/builtin_frames/icrs_cirs_transforms.py:15]: True, but then we'd need to change the isort settings in pyproject.toml and re-sort all the other sorted modules. So I don't think the total diff size will change much.

Comment (nstarman) [reply astropy/coordinates/builtin_frames/icrs_cirs_transforms.py:15]: When ``black`` is adopted ``isort`` will have to change and everything re-sorted in one go. The important thing here is that I'm making sure we don't and won't have import cycles.

File: astropy/coordinates/angle_formats.py (hunk 1 above)
Comment (mhvk) [reply astropy/coordinates/angle_formats.py:31]: Happy with that - nice to separate auto-reformats that could in fact change execution from those that are guaranteed not to.

File: astropy/coordinates/builtin_frames/icrs_cirs_transforms.py (hunk 2 above)
Comment (eerovaher) [reply astropy/coordinates/builtin_frames/icrs_cirs_transforms.py:15]: > True, but then we'd need to change the isort settings in pyproject.toml and re-sort all the other sorted modules.

Indeed better to do that when/if `black` is adopted.

**Review by eerovaher — APPROVED** [review]

**Pull request merged**